### ISSUE TITLE
Clarify guidelines on Bring-Your-Own-Storage

### ISSUE THREAD

**Issue by mikekistler** [Microsoft/api-guidelines#320]
Some teams report that the current language is confusing, e.g. "Is "Bring Your Own Storage" a design pattern or service/API/library?"

We should try to clarify that it is the _customer_ that creates and stores the data into an Azure Storage account and then just provides the location of the data to the API.

**Comment by mikekistler** [comment]
I think the way to handle this is to create a "Bring Your Own Storage" topic in the ConsiderationsForServiceDesign that provides the context and rationale for the guidelines.

**Comment by JeffreyRichter** [comment]
Yes, and maybe just a 1 or 2 sentence intro to the section in the guidelines too (in case people don't see the considerations section)?
